Spawn Summoned Voles Around Player and Add Second Summon Ring on Condition

Lukas-bit opened the pull request:
During the Behemoth Vole's second phase, the summoned Giant Voles will now spawn in a ring around the player. Also, if the boss is not already in its second phase, and its health is reduced to 30% or less, then a larger second ring of Giant Voles will spawn around the player in addition to the first one.

Lukas-bit committed:
Merge branch 'master' of https://github.com/noahwagner04/CSE320_Game

Lukas-bit committed:
Make Guard Voles Spawn in a Ring Around Player

Changed the summoned Giant Voles so that they will spawn
around the player instead of the Behemoth Vole. The summoned Giant Voles
are now siblings of the Behemoth Vole in the scene tree, instead of
children.

Lukas-bit committed:
Remove Accidental Plus Symbol & Unneeded Comments

Lukas-bit committed:
Merge branch 'lukas/behemoth_vole_summon_guards' into lukas/behemoth_vole

Lukas-bit committed:
Merge branch 'lukas/behemoth_vole' into lukas/behemoth_vole_summon_guards

Lukas-bit committed:
Create Larger Summon Ring On Condition

If the behemoth vole is brought down to 30% or below its max hp before
it spawns the first ring of voles, then a secondary larger ring of
Giant Voles will spawn around the player in addition to the smaller
first ring. Also, increased the damage behemoth can do by 5 points.

Lukas-bit committed:
Merge branch 'master' of https://github.com/noahwagner04/CSE320_Game

Lukas-bit committed:
Merge branch 'master' into lukas/behemoth_vole_summon_guards

Lukas-bit committed:
Remove Comments

Lukas-bit committed:
Change Value Passed to summon_voles in 30% Check

obimax99 reviewed (approved):
I think there could be a way to use math to eliminate the need for the "if (large_ring ==1)" section that would allow you to more easily add or remove the amount of voles but it's probably not worth it since we most likely won't change it. Looks good to me!

Lukas-bit committed:
Make Ring of Voles Amount More Variable

I liked Max's suggestion of using math to make adding or removing voles
easier, so I made the number of summoned voles determined by a for loop
nested within a while loop. This also makes the difficulty of the second
phase scale more off of the player's strength

Lukas-bit committed:
Delete Commented Code

Forgot to delete some commented out code from the previous version of this
feature

Lukas-bit committed:
Change Boss Aggro Distance on Condition

The boss will have a greater aggro distance when chasing. Otherwise,
it'll have the default of 300.

Lukas-bit committed:
Make Summoned Voles Spawn Around Boss

During the boss's second phase, the voles will summon in ring(s) around
the boss instead of a player
